File: php/LeadDatabase/Campaigns/ScheduleCampaign.php
<?php
/*
   ScheduleCampaign.php

   Marketo REST API Sample Code
   Copyright (C) 2016 Marketo, Inc.

   This software may be modified and distributed under the terms
   of the MIT license.  See the LICENSE file for details.
*/
$request = new ScheduleCampaign();
$request->id = 1001;
$token1 = new stdClass();
$token1->name = "{{my.token}}";
$token1->value = "Hello World!";
$request->tokens = array($token1);
print_r($request->postData());

class ScheduleCampaign{
	private $host = "CHANGE ME";
	private $clientId = "CHANGE ME";
	private $clientSecret = "CHANGE ME";
	public $id;//id of campaign to schedule
	public $runAt;//dateTime to run campaign
	public $cloneToProgramName;//if set will clone program with name
	public $tokens;//array of stdClass objects with two members, name and value
	
	public function postData(){
		$url = $this->host . "/rest/v1/campaigns/" . $this->id . "/schedule.json?access_token=" . $this->getToken();
		$ch = curl_init($url);
		$requestBody = $this->bodyBuilder();
		print_r($requestBody);
		curl_setopt($ch,  CURLOPT_RETURNTRANSFER, 1);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array('accept: application/json','Content-Type: application/json'));
		curl_setopt($ch, CURLOPT_POST, 1);
		curl_setopt($ch, CURLOPT_POSTFIELDS, $requestBody);
		curl_getinfo($ch);
		$response = curl_exec($ch);
		return $response;
	}
	
	private function getToken(){
		$ch = curl_init($this->host . "/identity/oauth/token?grant_type=client_credentials&client_id=" . $this->clientId . "&client_secret=" . $this->clientSecret);
		curl_setopt($ch,  CURLOPT_RETURNTRANSFER, 1);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array('accept: application/json',));
		$response = json_decode(curl_exec($ch));
		curl_close($ch);
		$token = $response->access_token;
		return $token;
	}
	private function bodyBuilder(){
		$body = new stdClass();
		$body->input = new stdClass();
		if (isset($this->runAt)){
			$body->input->runAt = $this->runAt;
		}
		if (isset($this->cloneToProgramName)){
			$body->input->cloneToProgramName = $this->cloneToProgramName;
		}
		if (isset($this->tokens)){
			$body->input->tokens = $this->tokens;
		}
		$json = json_encode($body);
		return $json;
	}	
}

File: c#/LeadDatabase/Campaigns/ScheduleCampaign.cs
/*
   ScheduleCampaign.cs

   Marketo REST API Sample Code
   Copyright (C) 2016 Marketo, Inc.

   This software may be modified and distributed under the terms
   of the MIT license.  See the LICENSE file for details.
*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;
using System.IO;
using System.Collections;
using Newtonsoft.Json;

namespace Samples
{

    class ScheduleCampaign
    {
        private String host = "CHANGE ME"; //host of your marketo instance, https://AAA-BBB-CCC.mktorest.com
        private String clientId = "CHANGE ME"; //clientId from admin > Launchpoint
        private String clientSecret = "CHANGE ME"; //clientSecret from admin > Launchpoint
        public int id;//id of campaign to schedule, required
        public String runAt; //datetime to schedule campaign for, 5 mins later if unspecified
        public String cloneToProgramName;//if specified will clone all content and parent program to a new program with the specified name
        public List<Dictionary<String, String>> tokens; //list of token dictionaries with members name and value

        /*
        public static void Main(string[] args)
        {
            ScheduleCampaign sched = new ScheduleCampaign();
            sched.id = 1573;
            sched.cloneToProgramName = "New API Test Program";
            List<Dictionary<string, string>> tokenList = new List<Dictionary<string, string>>();
            Dictionary<string, string> token1 = new Dictionary<string, string>();
            token1.Add("name", "{{my.token}}");
            token1.Add("value", "This token has been replaced");
            tokenList.Add(token1);
            String result = sched.postData();
            Console.WriteLine(result);
        }
        */


        public String postData()
        {
            //Assemble the URL
            String url = host + "/rest/v1/campaigns/" + id + "/schedule.json?access_token=" + getToken();
            //Serialize the Data into JSON
            String requestBody = bodyBuilder();
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
            request.Method = "POST";
            request.ContentType = "application/json";
            request.Accept = "application/json";
            StreamWriter wr = new StreamWriter(request.GetRequestStream());
            wr.Write(requestBody);
            wr.Flush();
            HttpWebResponse response = (HttpWebResponse)request.GetResponse();
            Stream resStream = response.GetResponseStream();
            StreamReader reader = new StreamReader(resStream);
            return reader.ReadToEnd();
        }
        private String bodyBuilder()
        {
            Dictionary<String, Object> parent = new Dictionary<string, object>();
            Dictionary<String, Object> input = new Dictionary<string,object>();
            if (runAt != null){
                input.Add("runAt", runAt);
            }
            if (cloneToProgramName != null){
                input.Add("cloneToProgramName", cloneToProgramName);
            }
            if (tokens != null){
                input.Add("tokens", tokens);
            }
            return JsonConvert.SerializeObject(parent);
        }

        private String getToken()
        {
            String url = host + "/identity/oauth/token?grant_type=client_credentials&client_id=" + clientId + "&client_secret=" + clientSecret;
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
            request.ContentType = "application/json";
            HttpWebResponse response = (HttpWebResponse)request.GetResponse();
            Stream resStream = response.GetResponseStream();
            StreamReader reader = new StreamReader(resStream);
            String json = reader.ReadToEnd();
            //Dictionary<String, Object> dict = JavaScriptSerializer.DeserializeObject(reader.ReadToEnd);
            Dictionary<String, String> dict = JsonConvert.DeserializeObject<Dictionary<String, String>>(json);
            return dict["access_token"];
        }
    }
}
